xrpl4j-core/src/test/java/org/xrpl/xrpl4j/crypto/TestConstants.java

```diff
@@ -36,18 +37,39 @@ public interface TestConstants {
   String ED_PUBLIC_KEY_B58 = "aKEusmsH9dJvjfeEg8XhDfpEgmhcK1epAtFJfAQbACndz5mUA73B";
   PublicKey ED_PUBLIC_KEY = PublicKey.fromBase16EncodedPublicKey(ED_PUBLIC_KEY_HEX);
 
-  String ED_PRIVATE_KEY_HEX = "EDB224AFDCCEC7AA4E245E35452585D4FBBE37519BCA3929578BFC5BBD4640E163";
-  String ED_PRIVATE_KEY_B58 = "pDcQTi2uFBAzQ7cY2mYQtk9QuQBoLU6rJypEf8EYPQoouh";
-  PrivateKey ED_PRIVATE_KEY = PrivateKey.of(UnsignedByteArray.of(BaseEncoding.base16().decode(ED_PRIVATE_KEY_HEX)));
+  String ED_PRIVATE_KEY_HEX = "B224AFDCCEC7AA4E245E35452585D4FBBE37519BCA3929578BFC5BBD4640E163";
+  String ED_PRIVATE_KEY_WITH_PREFIX_HEX = "ED" + ED_PRIVATE_KEY_HEX;
+
+  /**
+   * Helper method to return a newly constructed {@link PrivateKey} with its own copy of bytes.
+   *
+   * @return A {@link PrivateKey}.
+   */
+  static PrivateKey getEdPrivateKey() {
+    return PrivateKey.fromNaturalBytes(
+      UnsignedByteArray.of(BaseEncoding.base16().decode(ED_PRIVATE_KEY_HEX)),
+      KeyType.ED25519
+    );
+  }
 
   // Secp256k1 Public Key
   String EC_PUBLIC_KEY_HEX = "027535A4E90B2189CF9885563F45C4F454B3BFAB21930089C3878A9427B4D648D9";
   String EC_PUBLIC_KEY_B58 = "aB4ifx88a26RYRSSzeKW8HpbXfbpzQFRsX6dMNmMwEVHUTKzfWdk";
   PublicKey EC_PUBLIC_KEY = PublicKey.fromBase16EncodedPublicKey(EC_PUBLIC_KEY_HEX);
 
-  String EC_PRIVATE_KEY_HEX = "00DAD3C2B4BF921398932C889DE5335F89D90249355FC6FFB73F1256D2957F9F17";
-  String EC_PRIVATE_KEY_B58 = "rEjDwJp2Pm3NrUtcf8v17jWopvqPJxyi5RTrDfhcJcWSi";
-  PrivateKey EC_PRIVATE_KEY = PrivateKey.of(UnsignedByteArray.of(BaseEncoding.base16().decode(EC_PRIVATE_KEY_HEX)));
+  String EC_PRIVATE_KEY_HEX = "DAD3C2B4BF921398932C889DE5335F89D90249355FC6FFB73F1256D2957F9F17";
+  String EC_PRIVATE_KEY_WITH_PREFIX_HEX = "00" + EC_PRIVATE_KEY_HEX;
+
+  /**
+   * Helper method to return a newly constructed {@link PrivateKey} with its own copy of bytes.
+   *
+   * @return A {@link PrivateKey}.
+   */
+  static PrivateKey getEcPrivateKey() {
+    return PrivateKey.fromNaturalBytes(
+      UnsignedByteArray.of(BaseEncoding.base16().decode(EC_PRIVATE_KEY_HEX)), KeyType.SECP256K1
+    );
+  }
```

Comment:
We should revert this and keep using `EC_PRIVATE_KEY` in our tests to make sure we are always copying and can't mutate PrivateKey bytes.

Reply:
@nkramer44 and I decided to keep this code, but add unit tests to validate PrivateKey immutability, which was added via 46db60ceef1788f3e0646cea567eaf1a3298e413.